features/environment.py

```diff
@@ -68,9 +67,14 @@ def get_validation_outcome_hash(obj):
 def after_scenario(context, scenario):
     # Given steps may introduce an arbitrary amount of stackframes.
     # we need to clean them up before behave starts appending new ones.
+    old_outcomes = getattr(context, 'gherkin_outcomes', [])
     while context._stack[0].get('@layer') == 'attribute':
         context._pop()
+    # preserve the outcomes to be serialized to DB in after_feature()
+    context.gherkin_outcomes = old_outcomes
+
 
+def after_feature(context, feature):
```

Comment:
This is the main change. We don't want N/A results + X results in the dashboard at the feature level, but we're taking max severity at the scenario level.